### ISSUE TITLE
8$: Add JSON Schema for quick-lint-js.config files

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#927]
- [x] Create JSON schema (https://github.com/quick-lint/quick-lint-js/pull/936)
- [ ] Publish schema to `https://quick-lint-js.com/config/quick-lint-js-config-schema`
- [ ] (when shipping) Add version number
- [ ] Document how users can use the schema

**Comment by HarshitAghera** [comment]
@strager, can I take up this issue?

**Comment by HarshitAghera** [comment]
Should we create quick-lint-js.schema.json file and and parse each .cofig file against schema file to detect error, here?
simdjson doesn't have a schema support. How should I proceed?

https://github.com/quick-lint/quick-lint-js/blob/d43f1240d5cfe18684ca3590821aafe6294425a0/src/quick-lint-js/configuration/configuration.cpp#L73-L82

**Comment by strager** [comment]
> Should we create quick-lint-js.schema.json file and and parse each .cofig file against schema file to detect error, here?

No. We don't need to validate the schema ourselves because we already do that.

This task is to provide a schema so *others* can use the schema as documentation and for editor completion.

**Comment by amanmalh** [comment]
Hi @strager, can I take this up if there is no progress on this?

**Comment by strager** [comment]
@amanmalh Let's give @HarshitAghera some time to respond.

**Comment by HarshitAghera** [comment]
Sorry for the late reply. Feature implementation is almost completed. I will open PR soon for review.

**Comment by strager** [comment]
- [x] Create JSON schema (https://github.com/quick-lint/quick-lint-js/pull/936)
- [ ] Publish schema to `https://quick-lint-js.com/config/quick-lint-js-config-schema`
- [ ] (when shipping) Add version number
- [ ] Document how users can use the schema

@HarshitAghera I imagine the version number belongs in the schema URL too? Where would the version number go exactly? `https://quick-lint-js.com/config/quick-lint-js-config-schema-1.2.3`?

**Comment by HarshitAghera** [comment]
>@HarshitAghera I imagine the version number belongs in the schema URL too? Where would the version number go exactly? https://quick-lint-js.com/config/quick-lint-js-config-schema-1.2.3?

Yes, I think it seems the right way to do it.
